In [2]:
import pandas as pd
archivo = '..\data\ISBSG - Release May 2017 R1.csv'
#archivo = '..\data\ISBSG DATA Release 12.csv'
df = pd.read_csv(archivo, sep = ';', low_memory = False)
variables = ['Data Quality Rating', 'UFP rating', 'Industry Sector','Application Group', 'Development Type', 'Development Platform', 'Language Type', 'Primary Programming Language', 'Count Approach', 'Functional Size', 'Adjusted Function Points', 'Normalised Work Effort Level 1', 'Summary Work Effort', 'Project Elapsed Time', 'Business Area Type', '1st Data Base System', 'Used Methodology', 'Resource Level', 'Max Team Size', 'Average Team Size', 'Input count', 'Output count', 'Enquiry count', 'File count', 'Interface count', 'Agile Method Used']
df = df.loc[:, variables]
filtro = ((df['Data Quality Rating'] == 'A') | (df['Data Quality Rating'] == 'B')) & ((df['UFP rating'] == 'A') | (df['UFP rating'] == 'B')) 
df = df.loc[filtro, :]
filtro = (df['Normalised Work Effort Level 1'].notnull()) & (df['Normalised Work Effort Level 1'] == df['Summary Work Effort'])
df = df.loc[filtro, :]
filtro = df['Count Approach'] == 'IFPUG 4+'
df = df.loc[filtro, :]
variables = ['Industry Sector','Application Group', 'Development Type', 'Development Platform', 'Language Type', 'Primary Programming Language', 'Functional Size', 'Adjusted Function Points', 'Normalised Work Effort Level 1', 'Project Elapsed Time', 'Business Area Type', '1st Data Base System', 'Used Methodology', 'Max Team Size', 'Average Team Size', 'Input count', 'Output count', 'Enquiry count', 'File count', 'Interface count', 'Agile Method Used']
df = df.loc[:, variables]
df = df.dropna(axis=1, thresh=int(0.5*len(df)))
print(len(df))
df = df.dropna()
df

2183


Industry Sector     Application Group Development Type  \
7            Insurance  Business Application  New Development   
43          Government  Business Application  New Development   
68           Utilities  Business Application  New Development   
121      Manufacturing  Business Application  New Development   
140   Service Industry  Business Application  New Development   
...                ...                   ...              ...   
7876        Government  Business Application      Enhancement   
7891         Financial  Business Application  New Development   
7924         Financial  Business Application      Enhancement   
7928        Government  Business Application      Enhancement   
7934        Government  Business Application  New Development   

     Development Platform Language Type Primary Programming Language  \
7                      MF           3GL                         Java   
43                     MF           3GL                        COBOL   
68                     MF           3GL                         PL/I   
121                 Multi           4GL                          SQL   
140                    PC           3GL                           C#   
...                   ...           ...                          ...   
7876                   MF           3GL                        COBOL   
7891                Multi           4GL                 PowerBuilder   
7924                   MR           3GL                        COBOL   
7928                   MF           3GL                         PL/I   
7934                Multi           4GL                    Datastage   

      Functional Size Adjusted Function Points  \
7               620.0                      620   
43              243.0                      250   
68              163.0                      152   
121             702.0                      702   
140              90.0                      101   
...               ...                      ...   
7876             12.0                       12   
7891            478.0                      478   
7924            242.0                      213   
7928             96.0                       96   
7934           1048.0                     1048   

      Normalised Work Effort Level 1 Project Elapsed Time  \
7                            18160.0                  7,0   
43                            2924.0                  8,0   
68                             668.0                  9,0   
121                           4805.0                 12,3   
140                           2006.0                  5,0   
...                              ...                  ...   
7876                           129.0                  3,0   
7891                          2716.0                  5,8   
7924                          2185.0                  7,0   
7928                          1011.0                  6,0   
7934                          7722.0                 16,0   

     1st Data Base System Used Methodology  
7                IMS;DB2;       Don't Know  
43                   DB2;              Yes  
68                   DB2;              Yes  
121    Oracle,SQL Server;       Don't Know  
140   MS SQL Server 2005;              Yes  
...                   ...              ...  
7876                VSAM;              Yes  
7891              Oracle;       Don't Know  
7924                 RDB;               No  
7928                 DB2;              Yes  
7934                 DB2;              Yes  

[624 rows x 12 columns]

Nos quedan 624 proyectos aplicando un filtrado similar al original.